File: Thomas/src/main/java/cc/android/thomas/WorkoutSummaryAdapter.java
package cc.android.thomas;

import android.support.annotation.NonNull;
import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import java.util.ArrayList;
import java.util.List;

import cc.lib.game.Utils;

public class WorkoutSummaryAdapter extends RecyclerView.Adapter<WorkoutSummaryAdapter.Holder> {

    final List<Object[]> text = new ArrayList<>();

    WorkoutSummaryAdapter(MainActivity.Workout workout, int which) {
        text.add(Utils.toArray(R.string.workout_column_station, which));
        for (MainActivity.Station s : workout.stations) {
            if (s.enabled) {
                long secs = 0;
                switch (which) {
                    case 0: // today
                        secs = s.todaySecs;
                        break;
                    case 1: // this week
                        secs = s.thisWeekSecs;
                        break;
                    case 2: // this month
                        secs = s.thisMonthSecs;
                        break;
                    default: // all time
                        secs = s.allTimeSecs;
                        break;
                }
                if (secs > 0)
                    text.add(Utils.toArray(s.name, MainActivity.getTimeString(secs)));
            }
        }
    }

    @NonNull
    @Override
    public Holder onCreateViewHolder(@NonNull ViewGroup viewGroup, int i) {
        View view = LayoutInflater.from(viewGroup.getContext()).inflate(R.layout.workout_summary_list_item, viewGroup, false);
        return new Holder(view);
    }

    @Override
    public void onBindViewHolder(@NonNull Holder holder, int i) {
        Object [] o = text.get(i);
        if (o instanceof String[]) {
            holder.set((String)o[0], (String)o[1]);
        } else if (o instanceof Integer[]) {
            holder.set((Integer)o[0], (Integer)o[1]);
        }
    }

    @Override
    public int getItemCount() {
        return text.size();
    }

    public static class Holder extends RecyclerView.ViewHolder {

        TextView stationName, stationTime;

        public Holder(@NonNull View itemView) {
            super(itemView);
            stationName = itemView.findViewById(R.id.tv_station_name);
            stationTime = itemView.findViewById(R.id.tv_station_time);
        }

        void set(String name, String time) {
            stationName.setText(name);
            stationTime.setText(time);
        }

        void set(int stringResId, int stringArrayIndex) {
            stationName.setText(stringResId);
            stationTime.setText(stationTime.getResources().getStringArray(R.array.workout_column_names)[stringArrayIndex]);
        }
    }
}

File: Thomas/src/main/java/cc/android/thomas/WorkoutSummaryAdapter.kt
package cc.android.thomas

import android.content.Context
import android.view.LayoutInflater
import android.view.ViewGroup
import androidx.recyclerview.widget.RecyclerView
import cc.android.thomas.databinding.WorkoutSummaryListItemBinding
import cc.lib.utils.toHMS
import cc.lib.utils.weakReference
import java.util.*

data class Info(val str: String, val time: String)

class WorkoutSummaryAdapter internal constructor(val _context: Context, stations: Collection<MainActivity.Station>, which: Int) : RecyclerView.Adapter<WorkoutSummaryAdapter.Holder>() {
    val text: MutableList<Info> = ArrayList()
	val context by weakReference(_context)

    override fun onCreateViewHolder(viewGroup: ViewGroup, i: Int): Holder {
        return Holder(WorkoutSummaryListItemBinding.inflate(LayoutInflater.from(viewGroup.context), viewGroup, false))
    }

    override fun onBindViewHolder(holder: Holder, i: Int) {
        holder.set(text[i].str, text[i].time)
    }

    override fun getItemCount(): Int {
        return text.size
    }

	fun getTimeString(secs: Long): String = when (secs) {
		in 0..60 -> context?.getString(R.string.text_n_secs, secs)?:secs.toString()
		in 60..(60 * 60) -> String.format("%d:%02d", secs / 60, secs % 60)
		else -> secs.toHMS().let {
			String.format("%d:%02d:%02d", it[0], it[1], it[2])
		}
	}

	class Holder(var binding: WorkoutSummaryListItemBinding) : RecyclerView.ViewHolder(binding.root) {
        operator fun set(name: String, time: String) {
            binding.tvStationName.text = name
            binding.tvStationTime.text = time
        }
    }

    init {
        text.add(Info(_context.getString(R.string.workout_column_station), _context.resources.getStringArray(R.array.workout_column_names)[which]))
        stations.forEach { s->
            when (which) {
                0 -> s.todaySecs
                1 -> s.thisWeekSecs
                2 -> s.thisMonthSecs
                else -> s.allTimeSecs
            }.also { secs ->
	            if (secs > 0)
		            text.add(Info(s.name, getTimeString(secs)))
            }
        }
    }
}